src/systems.rs

```diff
@@ -47,6 +55,8 @@ pub fn update_action_state<A: Actionlike>(
     maybe_keyboard_input_stream: Option<Res<Input<KeyCode>>>,
     maybe_mouse_input_stream: Option<Res<Input<MouseButton>>>,
     clash_strategy: Res<ClashStrategy>,
+    mut res_as: Option<ResMut<ActionState<A>>>,
+    mut res_im: Option<ResMut<InputMap<A>>>,
```

Comment:
Why are you using `res_` prefix? It's obvious that this is a resource. I would use `input_map` and `action_state`.

Reply:
This was specifically to disambiguate the query from the resource. I had them as action_state_resource but it seemed too long. I can set them back to that if desired.

Reply:
Yeah, that's my preference here.